### PR TITLE
[FIX] allow searching for new style function

### PR DESCRIPTION
**Pull request by hbrunn**
fields in a search path

The current code breaks for domains like
`[('my_searchable_x2many_function_field.some_field', '=', 'some_value')]`

### CODE REVIEW

File: openerp/osv/expression.py [hunk 1]
```diff
@@ -866,7 +866,16 @@ def push_result(leaf):
                     domain = []
                 else:
                     # Let the field generate a domain.
-                    recs = model.browse(cr, uid, [], context)
+                    if len(path) > 1:
+                        recs = comodel.browse(
```
Comment (pedrobaeza) [inline openerp/osv/expression.py]: Why you need a browse? You have already ids list in the search return.

Comment (hbrunn) [inline openerp/osv/expression.py]: yes, why? Thanks!

**Comment by pedrobaeza** [comment]
Checked today that the patch fix the issue.

:+1:

**Comment by hbrunn** [comment]
@rco-odoo I just added a test for this

**Comment by antonylesuisse** [comment]
Closing this PR, i dont understand the patch.

**Comment by pedrobaeza** [comment]
@antonylesuisse, the PR is a bit dirty, but the goal is clear in the first PR's comment, and it includes a test for proving the failing. Can you give an opportunity to @hbrunn to fix the PR? @hbrunn, can you please leave 1 or 2 commits (if you put the test in a separate commit) for your patch?

**Comment by antonylesuisse** [comment]
Push force an updated version and reopen the PR.

**Comment by hbrunn** [comment]
@antonylesuisse lacking permissions to reopen, I created a new PR at https://github.com/odoo/odoo/pull/10228

**Pull request closed without merging**